validphys2/src/validphys/commondataparser.py

```diff
@@ -1,18 +1,319 @@
+into useful structures that can be fed to the main :py:class:`validphys.coredata.CommonData` class.
+
+A CommonData file is completely defined by a name (which defines the folder in which the information is)
+and a variant (which defines which files inside of the folder will be read).
+
+
+In this module a few auxiliary dataclasses that hold special information:
+    - TheoryMeta: contains the necessary information to read the (new style) fktables
+    - KinematicsMeta: containins metadata about the kinematics
+    - ReferenceMeta: literature references for the dataset
+
+The CommonMetaData defines how the CommonData file is to be loaded,
+by modifying the CommonMetaData using one of the loaded Variants one can change the resulting
+:py:class:`validphys.coredata.CommonData` object.
 """
-from operator import attrgetter
+from copy import copy
+from functools import cached_property
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Optional
 
 import pandas as pd
+from reportengine.compat import yaml
+from validobj.custom import Parser
+from validobj import ValidationError, parse_input
 
-from validphys.core import peek_commondata_metadata
+from validphys.utils import parse_yaml_inp
 from validphys.coredata import CommonData
 
+EXT = "pineappl.lz4"
+
+
+# Auxiliary parser for common types or sanity checks
+@Parser
+def ValidPath(path_str: str) -> Path:
+    """Parse strings into paths"""
+    return Path(path_str)
+
+
+@Parser
+def ValidOperation(op_str: str) -> str:
+    """Ensures that the operation defined in the commondata file is implemented in validphys"""
+    ret = op_str.upper()
+    # TODO: move accepted operations to this module so that the convolution receives an operation to apply
+    # instead of an operation to understand
+    from validphys.convolution import OP
+
+    if ret not in OP:
+        raise ValidationError(f"The operation '{op_str}' is not implemented in validphys")
+    return ret
+
+
+# Auxiliary objects
+@dataclass
+class ApfelComb:
+    """Some of the grids might have been converted from apfelcomb and introduce hacks.
+    These are the allowed hacks:
+        - repetition_flag:
+            list of fktables which might need to be repeated
+            necessary to apply c-factors in compound observables
+        - normalization:
+            mapping with the single fktables which need to be normalized and the factor
+            note that when they are global factors they are promoted to conversion_factor
+        - shifts:
+            mapping with the single fktables and their respective shifts
+            necessary to create "gaps" that some cfactors or datasets might expect
+    """
+
+    repetition_flag: Optional[list[str]] = None
+    normalization: Optional[dict] = None
+    shifts: Optional[dict] = None
+
+    @classmethod
+    def parser(cls, meta: dict):
+        return parse_input(meta, cls)
+
+ValidApfelComb = Parser(ApfelComb.parser)
```

Comment:
Why do we need this? Can it not be apfelcomb?

Reply:
I think when I tried it didn't work? I'll try again and copy here the error, maybe I'm doing something wrong.

Reply:
Without the `.parser` I get the following error:

```
ValidationError: Problem processing key at line 50 in .../metadata.yaml:
Cannot process field 'theory' of value into the corresponding field of 'CommonMetaData'
Problem processing key at line 51 in .../metadata.yaml:
Cannot process field 'apfelcomb' of value into the corresponding field of 'TheoryMeta'
No match for any possible type:
Not a valid match for 'Annotated': No match for any possible type:
Not a valid match for 'list': Expecting value to be a list, not CommentedMap
Not a valid match for 'NoneType': Expecting value of type 'NoneType', not CommentedMap.
Not a valid match for 'NoneType': Expecting value of type 'NoneType', not CommentedMap.
```

so I need the explicit call to `validobj.parse_input`.